src/models/capture_group_patterns.rs

```diff
@@ -56,3 +66,50 @@ impl Instantiate for CGPattern {
     CGPattern::new(self.pattern().instantiate(&substitutions))
   }
 }
+
+#[derive(Debug)]
+pub(crate) enum CompiledCGPattern {
+  Q(Query),
+  R(Regex), // Regex is not yet supported
+}
+
+impl CompiledCGPattern {
+  /// Applies the query upon the given node, and gets all the matches
```

Comment:
Is the documentation correct here? It seems this returns 0 or 1 matches. Is this meant to be the docstring for `get_matches`?